### PR TITLE
PWN-3253 - transaction details bottom sheet implementation

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-3253

## Description of Work 

Transaction details bottom sheet implementation

## Screenshot

image

### CODE REVIEW

**Comment by kuznetsov** [comment]
Please remove template text from PR description

**Comment by kuznetsov** [comment]
The screenshot in description may be useful 🙂

**Review by kuznetsov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/history/ui/detailsbottomsheet/TransactionDetailsBottomSheetPresenter.kt [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+                    view?.showError(R.string.error_general_message)
+                }
+            } catch (e: Throwable) {
+                Timber.e(e, "Error loading transaction details")
+                view?.showError(R.string.details_transaction_not_found)
+            } finally {
+                view?.showLoading(false)
+            }
+        }
+    }
```
[the remark this answers is not in the document]
Comment (1sakov) [reply app/src/main/java/org/p2p/wallet/history/ui/detailsbottomsheet/TransactionDetailsBottomSheetPresenter.kt:54]: That's absolutely okay. 
can't see any hell

File: app/src/main/res/values/styles_fonts.xml [hunk 2]
```diff
@@ -94,16 +95,21 @@
         <item name="android:textSize">12sp</item>
     </style>
 
-    <style name="WalletTheme.TextAppearance.Medium16">
-        <item name="android:textColor">@color/textIconPrimary</item>
-        <item name="fontFamily">@font/manrope_medium</item>
+    <style name="WalletTheme.TextAppearance.Medium20" parent="WalletTheme.TextAppearance.Medium">
+        <item name="android:textSize">20sp</item>
```
Comment (1sakov) [inline app/src/main/res/values/styles_fonts.xml:99]: Why did you remove font here and below?

File: app/src/main/java/org/p2p/wallet/history/ui/detailsbottomsheet/TransactionDetailsBottomSheetFragment.kt [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    override fun getTheme(): Int = R.style.WalletTheme_BottomSheet_Rounded
+
+    override fun onStart() {
+        super.onStart()
+        BottomSheetBehavior.from(requireView().parent as View).apply {
+            state = BottomSheetBehavior.STATE_EXPANDED
+        }
+    }
+
+    override fun showError(messageId: Int) {
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/history/ui/detailsbottomsheet/TransactionDetailsBottomSheetFragment.kt:63]: `@StringRes`

File: app/src/main/res/layout/dialog_transaction_details.xml [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+    android:layout_height="match_parent"
+    android:orientation="vertical">
+
+    <androidx.core.widget.NestedScrollView
+        android:id="@+id/scrollView"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:background="@drawable/bg_alert_elevation_rounded_top"
+        android:scrollbars="none"
+        app:layout_behavior="com.google.android.material.appbar.AppBarLayout$ScrollingViewBehavior">
```
Comment (1sakov) [inline app/src/main/res/layout/dialog_transaction_details.xml:16]: Why do you need this behavior here?

**Review by 1sakov — CHANGES_REQUESTED** [review]

File: app/src/main/res/values/styles_fonts.xml (hunk 2 above)
Comment (kabramovp2p) [reply app/src/main/res/values/styles_fonts.xml:99]: i moved it to parent -> WalletTheme.TextAppearance.Medium

File: app/src/main/res/layout/dialog_transaction_details.xml (hunk 4 above)
Comment (kabramovp2p) [reply app/src/main/res/layout/dialog_transaction_details.xml:16]: i think no need will be removed

File: app/src/main/java/org/p2p/wallet/history/ui/details/TransactionDetailsPresenter.kt [hunk 5]
```diff
@@ -163,7 +168,11 @@ class TransactionDetailsPresenter(
 
         val usdTotal = "(${transaction.getFormattedAmount()})"
         val total = "${transaction.getFormattedTotal()} $usdTotal"
-        val amount = SpanUtils.highlightText(total, usdTotal, context.getColor(R.color.textIconSecondary))
+        val amount = SpanUtils.highlightText(
+            total,
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/history/ui/details/TransactionDetailsPresenter.kt:172]: do you mind adding named args here?

File: app/src/main/res/values/styles_fonts.xml (hunk 2 above)
Comment (1sakov) [reply app/src/main/res/values/styles_fonts.xml:99]: oh okay

**Review by 1sakov — APPROVED** [review]
